File: python/arrays/lists/functions_operating_on_lists/operations_on_elements/insert.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

from pprint import pprint

list = [2, 4, 6]

print("Initial state:\n")
pprint(list); print()

r = list.insert(1, 8)

print("After add by list.insert(1, 8):\n")
pprint(list); print()
print(f"Returned value: {r}\n\n")

r = list.insert(4, 10)

print("After add by list.insert(4, 10):\n")
pprint(list); print()
print(f"Returned value: {r}\n\n")

r = list.insert(10, 12)

print("After add by list.insert(10, 12):\n")
pprint(list); print()
print(f"Returned value: {r}\n\n")

r = list.insert(0, 14)

print("After add by list.insert(0, 14):\n")
pprint(list); print()
print(f"Returned value: {r}\n\n")

r = list.insert(-1, 16)

print("After add by list.insert(-1, 16):\n")
pprint(list); print()
print(f"Returned value: {r}\n\n")

File: ruby/arrays/arrays/functions_operating_on_arrays/operations_on_elements/insert.rb
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md

array = [2, 4, 6]

puts("Initial state:\n\n")
puts(array.inspect, "\n")

r = array.insert(1, 8)

puts("After insert by array.insert(1, 8):\n\n")
puts(array.inspect, "\n")
puts("Returned value: #{r}\n\n")

r = array.insert(4, 10)

puts("After insert by array.insert(4, 10):\n\n")
puts(array.inspect, "\n")
puts("Returned value: #{r}\n\n")

r = array.insert(10, 12)

puts("After insert by array.insert(10, 12):\n\n")
puts(array.inspect, "\n")
puts("Returned value: #{r}\n\n")

r = array.insert(0, 14)

puts("After insert by array.insert(0, 14):\n\n")
puts(array.inspect, "\n")
puts("Returned value: #{r}\n\n")

r = array.insert(-1, 16)

puts("After insert by array.insert(-1, 16):\n\n")
puts(array.inspect, "\n")
puts("Returned value: #{r}\n\n")
